app/views/components/inputDate.scala.html

```diff
@@ -0,0 +1,77 @@
+@*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@this(govukDateInput: GovukDateInput)
+
+@(
+  form: Form[_],
+  legendContent: String,
+  legendClasses: String = "govuk-fieldset__legend--xl",
+  id: String = "date",
+  hintText: Option[String] = None,
+  legendHidden: Boolean = false,
+  legendAsPageHeading: Boolean = false)(implicit messages: Messages)
+
+@govukDateInput(DateInput(
+    fieldset = Some(Fieldset(
+        legend = Some(
+            Legend(
+                content = Text(messages(legendContent)),
+                isPageHeading = legendAsPageHeading,
+                classes = if (legendAsPageHeading) "govuk-label--l" else if (legendHidden) "govuk-visually-hidden" else ""
+            )
+        )
+    )),
+    id = id,
+    items = Seq(
+        InputItem(
+            id = s"$id.day",
+            classes = s"govuk-input--width-2${if(componentHasError("day") || form.errors.exists(_.key.isEmpty)) " govuk-input--error" else ""}",
```

Comment:
form.errors.exists(_.key.isEmpty) is done three times, could do it once and make use of a variable I believe which would cut down the computation from 3 actions to 1

Reply:
I have removed these as I believe they are redundant now.